src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/ccd/callback/PreSubmitCallbackResponse.java

```diff
@@ -10,11 +10,15 @@
 
 public class PreSubmitCallbackResponse<T extends CaseData> {
 
-    private final T data;
-    private final Set<String> errors = new LinkedHashSet<>();
+    private T data;
+    private Set<String> errors = new LinkedHashSet<>();
```

Comment:
Just curious about the use a LinkedHashSet here as apposed to standard hashset? And why it went from from final to non-final?

Reply:
LinkedHashSet preserves insert order on iteration, whereas standard HashSet does not -- so the errors should be displayed in the order they were added.

As for the final -- i cannot remember, but what did change about this class is we have started to deserialize it, not just serialize it. I was of the view that final fields and deserialization are not things that work well together:
https://docs.oracle.com/javase/specs/jls/se7/html/jls-17.html#jls-17.5.3

Reply:
Yeah my mistake, I was thinking that ordering would be lost after making the copy, just read the docs.